runtime/sema/gen/main.go

```diff
@@ -1612,9 +1791,7 @@ func gen(inPath string, outFile *os.File, registerTypes bool) {
 		_ = ast.AcceptDeclaration[struct{}](declaration, &gen)
 	}
 
-	if registerTypes {
-		gen.generateTypeInit(program)
-	}
```

Comment:
Does this no longer create conflicts/duplicate type declarations?

Reply:
As far as I could see, previously the init code was always generated when the tool was run https://github.com/onflow/cadence/pull/2648/files/8c56d2d3f8960d7be25dc1224aacceb63a8c03a3#diff-9b1bcb947820865585b33db840df55c910a9e4edda99592b8983958d14998112L1658-L1661, but was never generated for the test cases https://github.com/onflow/cadence/pull/2648/files/8c56d2d3f8960d7be25dc1224aacceb63a8c03a3#diff-5758071b202796a3a1b0e6809ead97950514807dc7950b646160a23d3803498cL53-L56. 

I felt like it's better to always test all code that gets generated. When I was working on the code generation part for the init function, a run of the tests passed, but then a run of the tool crashed. It turned out that I had made some mistakes, but those weren't encountered during the tests.

Reply:
Go ignores directories named `testdata`, see https://pkg.go.dev/cmd/go/internal/test#pkg-variables, the files in the directory are not checked or compiled, so there shouldn't be any problems, like conflicts, because even the `.golden.go` files are basically just "plain-text files"